yt/frontends/gadget/data_structures.py

```diff
@@ -629,6 +629,15 @@ def _get_hvals(self):
         if "Parameters" in handle:
             hvals.update((str(k), v) for k, v in handle["/Parameters"].attrs.items())
         handle.close()
+
+        # ensure that 1-element arrays are reduced to scalars
+        updated_hvals = {}
+        for hvalname, value in hvals.items():
+            if isinstance(value, np.ndarray) and value.size == 1:
+                mylog.info(f"Reducing singleton array {hvalname} to scalar.")
```

Comment:
```suggestion
                mylog.info("Reducing single-element array %s to scalar.", hvalname)
```

Reply:
This is a violation of rule `G004`, `ruff` is supposed to flag it (I guess that's a bug that it doesn't)